### ISSUE TITLE
Including direction in setVisibleCoordinates calculates the viewport incorrectly

### ISSUE THREAD

**Issue by TimWatson** [mapbox/mapbox-gl-native#12075]
If you include direction when setting the visible coordinates it calculates the viewport first and then rotates the map resulting in an incorrect viewport.

Example:
```
var coordinates = [CLLocationCoordinate2D(latitude: 34.407548, longitude: -119.8696609),
							   CLLocationCoordinate2D(latitude: 34.4211754, longitude: -119.8394465)]
let padding = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
let direction = CLLocationDirection(90)
let animation = CAMediaTimingFunction(name: kCAMediaTimingFunctionEaseInEaseOut)
mapView.setVisibleCoordinates(&coordinates, count: UInt(coordinates.count), edgePadding: padding, direction: direction, duration: 0.4, animationTimingFunction: animation, completionHandler: nil)

```

iOS 11.4
Mapbox 4.0.1, 4.0.2

https://github.com/mapbox/mapbox-gl-native/blob/43efd61813dfddb0ac673314350d95aa1a86b8cb/platform/ios/src/MGLMapView.h#L762

**Comment by julianrex** [comment]
Thanks for the report @TimWatson

**Comment by TimWatson** [comment]
lol what. Does that mean this isn't getting fixed?

**Comment by julianrex** [comment]
Still on our list!

**Comment by TimWatson** [comment]
Hi Fabian, thanks for getting back to me. I need both the direction and the visible bounds to change but when using this function it currently calculates the new map bounds to fit the required visible bounds and then it applies the rotation. That means that the calculation is now wrong since the required visible bounds now fit differently in the new, rotated, map viewport. Your suggestion doesn't solve this issue.

**Comment by fabian-guerra** [comment]
Fixed in #13761
